src/core/server/saved_objects/migrations/actions/initialize_action.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import * as TaskEither from 'fp-ts/lib/TaskEither';
+import * as Either from 'fp-ts/lib/Either';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { ElasticsearchClient } from '../../../elasticsearch';
+import {
+  catchRetryableEsClientErrors,
+  RetryableEsClientError,
+} from './catch_retryable_es_client_errors';
+
+import { FetchIndexResponse, fetchIndices } from './fetch_indices';
+
+const routingAllocationEnable = 'cluster.routing.allocation.enable';
+export interface ClusterRoutingAllocationEnabled {
+  clusterRoutingAllocationEnabled: boolean;
+}
+
+export interface InitActionParams {
+  client: ElasticsearchClient;
+  indices: string[];
+}
+
+export interface ClusterAllocationDisabledError {
+  type: 'cluster_routing_allocation_disabled';
+}
+
+export const checkClusterRoutingAllocationEnabledTask =
+  ({
+    client,
+  }: {
+    client: ElasticsearchClient;
+  }): TaskEither.TaskEither<RetryableEsClientError | ClusterAllocationDisabledError, {}> =>
+  () => {
+    return client.cluster
+      .getSettings({
+        flat_settings: true,
+      })
+      .then((settings) => {
+        const clusterRoutingAllocations: string[] =
+          settings?.transient?.[routingAllocationEnable] ??
+          settings?.persistent?.[routingAllocationEnable] ??
+          [];
+
+        const clusterRoutingAllocationEnabled =
+          [...clusterRoutingAllocations].length === 0 ||
+          [...clusterRoutingAllocations].every((s: string) => s === 'all'); // if set, only allow 'all'
```

Comment:
Only allow `cluster.routing.allocation.enabled=all` as a persistent or transient setting if it is set.